### ISSUE TITLE
false positive in `prefer_asserts_in_initializer_lists`

### ISSUE THREAD

**Issue by kevmoo** [dart-lang/linter#1655]
This should be fine, but is not. Seems a lot like https://github.com/dart-lang/linter/issues/1557 (only comes up if there is a type parameter)

```dart
abstract class H<T> {
  H() {
    assert(_assert()); // OK
  }

  bool _assert() => true;
}
```

Dart VM version: 2.5.0-dev.1.0 (Tue Jul 9 15:27:01 2019 +0200) on "macos_x64"

**Comment by kevmoo** [comment]
```
info: Prefer putting asserts in initializer list. (prefer_asserts_in_initializer_lists at [linter] test/rules/prefer_asserts_in_initializer_lists.dart:130)
```

**Comment by kevmoo** [comment]
CC @pq
